packages/components/src/Layout/Semantics/Header.tsx

```diff
@@ -41,8 +42,11 @@ export const headerFooterCSS = css`
   flex: 0 0 auto;
 `
 
-export const Header = styled.header.withConfig({
-  shouldForwardProp,
-})<HeaderProps>`
+export const Header = styled.header
+  .withConfig({ shouldForwardProp })
+  .attrs<HeaderProps>(({ border }) => ({
+    border,
+  }))<HeaderProps>`
```

Comment:
`attrs` addition shoudn't be needed here

```suggestion
  <HeaderProps>`
```